server/service/integration_mdm_lifecycle_test.go

```diff
@@ -796,20 +821,28 @@ func (s *integrationMDMTestSuite) TestLifecycleSCEPCertExpiration() {
 	require.NoError(t, err)
 	require.Nil(t, cmd)
 
+	fmt.Println(">>> RE-ENROLL FOR RENEWAL")
+
 	// devices renew their SCEP cert by re-enrolling.
 	require.NoError(t, manualEnrolledDevice.Enroll())
 	require.NoError(t, automaticEnrolledDevice.Enroll())
 	require.NoError(t, automaticEnrolledDeviceWithRef.Enroll())
 	require.NoError(t, migratedDevice.Enroll())
 
+	s.runWorker()
```

Comment:
Because those "Enroll" calls were after the SCEP renewal thing, they should all have been seen as SCEPRenewalInProgress here: https://github.com/fleetdm/fleet/blob/56054eb63bd443d8b67a6742e50269f2b4af7cd0/server/service/apple_mdm.go#L2684-L2688

And so the TurnOn lifecycle event should not have been called, and the resulting command to install fleetd should not have been enqueued by the worker run (AIUI). But that's not what happens, there's always a couple hosts that are not seen as "in SCEP renewal", and it doesn't seem deterministic (not always the same).